src/lib/formatter/RuleResultRecombinator.ts

```diff
@@ -28,33 +28,46 @@ export class RuleResultRecombinator {
 				formattedResults = this.constructJunit(results);
 				break;
 			case OUTPUT_FORMAT.SARIF:
-				formattedResults = await constructSarif(results, executedEngines);
+				formattedResults = await constructSarif(results, executedEngines, outputOptions.normalizeSeverity);//
 				break;
 			case OUTPUT_FORMAT.TABLE:
 				formattedResults = this.constructTable(results);
 				break;
 			case OUTPUT_FORMAT.XML:
-				formattedResults = this.constructXml(results);
+				formattedResults = this.constructXml(results, outputOptions.normalizeSeverity);
 				break;
 			default:
 				throw new SfdxError('Unrecognized output format.');
 		}
-		return {minSev: this.findMinSev(results), results: formattedResults, summaryMap: this.generateSummaryMap(results, executedEngines)};
+		return {minSev: this.findMinSev(results, outputOptions.normalizeSeverity), results: formattedResults, summaryMap: this.generateSummaryMap(results, executedEngines)};
 	}
 
-	private static findMinSev(results: RuleResult[]): number {
+	private static findMinSev(results: RuleResult[], normalizeSeverity: boolean): number {
 		// If there are no results, then there are no errors.
 		if (!results || results.length === 0) {
 			return 0;
 		}
 		let minSev = null;
-		for (const res of results) {
-			for (const violation of res.violations) {
-				if (!minSev || violation.severity < minSev) {
-					minSev = violation.severity;
+
+		// if -n or -s flag used, minSev is calculated with normal value
+		if (normalizeSeverity) {
+			for (const res of results) {
+				for (const violation of res.violations) {
+					if (!minSev || violation.severity < minSev) {
```

Comment:
Shouldn't the comparison be against `violation.normalizedSeverity`?

We can reduce the duplication a little more by using a variable for the severity value and extracting the severity value right before comparison. Something like:
```
for (const res of results) {
  for (const violation of violations) {
    var severity = (normalizeSeverity)? violation.normalizedSeverity : violation.severity;
    if (!minSev || severity < minSev) {
      minSev = severity;
    }
  }
}
```

Reply:
Thinking about this some more, if we were to check for the presence of `violation.normalizedSeverity`, we'd know if `normalizeSeverity` is true. This way, we don't even have to pass around this value and make a decision wherever we need to. What do you think?

Reply:
I think that would definitely be cleaner that passing normalizeSeverity in. Should I do this everywhere? For example, I created the OutputOptions type so we could pass the boolean into the runRulesMatchingCriteria method which further passes it around.